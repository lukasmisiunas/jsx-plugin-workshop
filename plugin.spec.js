const babel = require('@babel/core');
const plugin = require('./plugin');

describe('parse-jsx-plugin', () => {
  it('Task 1: should transpile empty jsx tag', () => {
    const source = '<div></div>';
    const { code } = babel.transform(source, { plugins: [plugin] }) ?? {};
    expect(code).toMatchSnapshot();
  });

  it.skip('Task 2: should transpile string props', () => {
    const source = '<div foo="bar"></div>';
    const { code } = babel.transform(source, { plugins: [plugin] }) ?? {};
    expect(code).toMatchSnapshot();
  });

  it.skip('Task 3: should transpile expression props', () => {
    const source = '<div prop1={1} prop2={1 + 1}></div>';
    const { code } = babel.transform(source, { plugins: [plugin] }) ?? {};
    expect(code).toMatchSnapshot();
  });

  it.skip('Task 4: should transpile text children', () => {
    const source = '<div>Hello</div>';
    const { code } = babel.transform(source, { plugins: [plugin] }) ?? {};
    expect(code).toMatchSnapshot();
  });

  it.skip('Task 5: should transpile element children', () => {
    const source = '<div><span>Hello</span></div>';
    const { code } = babel.transform(source, { plugins: [plugin] }) ?? {};
    expect(code).toMatchSnapshot();
  });
});
