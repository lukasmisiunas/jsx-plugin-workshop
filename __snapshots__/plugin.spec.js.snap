// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse-jsx-plugin Task 1: should transpile empty jsx tag 1`] = `"React.createElement("div", {});"`;

exports[`parse-jsx-plugin Task 2: should transpile string props 1`] = `
"React.createElement("div", {
  foo: "bar"
});"
`;

exports[`parse-jsx-plugin Task 3: should transpile expression props 1`] = `
"React.createElement("div", {
  prop1: 1,
  prop2: 1 + 1
});"
`;

exports[`parse-jsx-plugin Task 4: should transpile text children 1`] = `"React.createElement("div", {}, "Hello");"`;

exports[`parse-jsx-plugin Task 5: should transpile element children 1`] = `"React.createElement("div", {}, React.createElement("span", {}, "Hello"));"`;
